sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -274,6 +274,76 @@ case class AlterTableUnsetPropertiesCommand(
 
 }
 
+
+/**
+ * A command to change the columns for a table, only support changing the comments of non-partition
+ * columns for now.
+ *
+ * The syntax of using this command in SQL is:
+ * {{{
+ *   ALTER TABLE table_identifier
+ *   CHANGE [COLUMN] column_old_name column_new_name column_dataType [COMMENT column_comment]
+ *   [FIRST | AFTER column_name];
+ * }}}
+ */
+case class AlterTableChangeColumnsCommand(
+    tableName: TableIdentifier,
+    columns: Map[String, StructField]) extends RunnableCommand {
+
+  // TODO: support change column name/dataType/metadata/position.
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val catalog = sparkSession.sessionState.catalog
+    val table = catalog.getTableMetadata(tableName)
+    val resolver = sparkSession.sessionState.conf.resolver
+    DDLUtils.verifyAlterTableType(catalog, table, isView = false)
+
+    // Create a map that converts the origin column to the new column with changed comment, throw
+    // a Exception if the column reference is invalid or the column name/dataType is changed.
+    val columnsMap = columns.map { case (oldName: String, newField: StructField) =>
+      // Find the origin column from schema by column name.
+      val originColumn = findColumn(table.schema, oldName, resolver)
+      // Throw a Exception if the column name/dataType is changed.
```

Comment:
It is not very common to use `Exception` here. Or you can change it to -> `an AnalysisException`.